### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by Hliebov**

### CODE REVIEW

File: utils/streams.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+(function () {
```
Comment (opasno) [inline utils/streams.js]: You don't need this wrapper in Node environment.

File: utils/streams.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	let filePath = argv[FILE] || argv[FILE[0]];
+	let cssPath = argv[PATH] || argv[PATH[0]];
+
+	if (keys.indexOf(HELP) === 1 || keys.indexOf(HELP[0]) === 1) {
+		printHelpMessage();
+		return;
+	}
+
+	if (keys.length < 1) {
+		printHelpMessage();
```
Comment (opasno) [inline utils/streams.js]: No need for `return`?

File: utils/streams.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+				console.log('JSON successfully created!')
+			})
+	}
+
+	function transformToJson(path) {
+		fs.createReadStream(path)
+			.pipe(through(function (chunk, enc, callback) {
+				csv()
+					.fromString(chunk.toString())
+					.on('json',(json)=>{
```
Comment (opasno) [inline utils/streams.js]: Consider using two-whitespace indentation - the current makes it hard to read a nested code.

File: utils/streams.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+						this.push(JSON.stringify(json));
+					});
+
+				callback()
```
Comment (opasno) [inline utils/streams.js]: Shouldn't we attach `callback` as a listener to the `"end"` event of `csv()` stream?

File: utils/streams.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+			done();
+		}
+
+		process.stdin
+			.pipe(stream)
+			.pipe(process.stdout);
+	}
+
+	function printHelpMessage() {
+		console.log('Available commands: --file (-f) --action (-a) --help (-h). ' +
```
Comment (opasno) [inline utils/streams.js]: You can use ES6 interpolation instead of `'a' + 'b'`

File: utils/streams.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+			'Actions: "io", "transform-uppercase", "transform-csv", "transform-csv-file", "bundle-css" ' +
+			'Has to have a least one argument!');
+	}
+
+	function bundleCss(directory) {
+		fs.readdir(directory, (err, files) => {
+			if (err) {
+				console.error(err);
+			}
```
Comment (opasno) [inline utils/streams.js]: If an error occured, `return`

File: utils/streams.js [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	function printHelpMessage() {
+		console.log('Available commands: --file (-f) --action (-a) --help (-h). ' +
+			'Actions: "io", "transform-uppercase", "transform-csv", "transform-csv-file", "bundle-css" ' +
+			'Has to have a least one argument!');
+	}
+
+	function bundleCss(directory) {
+		fs.readdir(directory, (err, files) => {
```
Comment (opasno) [inline utils/streams.js]: You should check for `.css` extension.

File: utils/streams.js [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+	const request = require('request');
+
+	const HELP = 'help';
+	const ACTION ='action';
+	const FILE ='file';
+	const PATH ='path';
+
+	const CSS_PATH = 'https://www.epam.com/etc/clientlibs/foundation/main.min.fc69c13add6eae57cd247a91c7e26a15.css';
+
+	let argv = require('minimist')(process.argv);
```
Comment (opasno) [inline utils/streams.js]: Consider configuring aliases via `alias` option of `minimist`.

File: utils/streams.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+			'Has to have a least one argument!');
+	}
+
+	function bundleCss(directory) {
+		fs.readdir(directory, (err, files) => {
+			if (err) {
+				console.error(err);
+			}
+			files = files.map(file => path.join(directory, file));
+			async.map(files, fs.readFile, (err, results) => {
```
Comment (opasno) [inline utils/streams.js]: Consider implementing it with the streams.

File: utils/streams.js (hunk 1 above)
Comment (Hliebov) [reply utils/streams.js]: agree

File: utils/streams.js (hunk 2 above)
Comment (Hliebov) [reply utils/streams.js]: yep, need one

File: utils/streams.js (hunk 4 above)
Comment (Hliebov) [reply utils/streams.js]: you're right)

File: utils/streams.js (hunk 5 above)
Comment (Hliebov) [reply utils/streams.js]: sure

File: utils/streams.js (hunk 6 above)
Comment (Hliebov) [reply utils/streams.js]: agree, thanks!

File: utils/streams.js (hunk 7 above)
Comment (Hliebov) [reply utils/streams.js]: done

File: utils/streams.js (hunk 8 above)
Comment (Hliebov) [reply utils/streams.js]: done

**Pull request closed without merging**